src/sentry/tasks/relay.py

```diff
@@ -47,71 +47,55 @@ def update_config_cache(
     sentry_sdk.set_tag("update_reason", update_reason)
     sentry_sdk.set_tag("generate", generate)
 
-    if not should_update_cache(
-        organization_id=organization_id, project_id=project_id, public_key=public_key
-    ):
```

Comment:
Setting the debounce key before enqueuing the task is not transactional and leaves a race condition in which multiple tasks could be scheduled. For now, we accept these duplicates and don't make further attempts to deduplicate them.